src/prefect/tasks/kubernetes/job.py

```diff
@@ -571,3 +573,134 @@ def run(
+        namespace: str = "default",
+        kube_kwargs: dict = None,
+        kubernetes_api_key_secret: str = "KUBERNETES_API_KEY",
+        **kwargs: Any,
+    ):
+        self.body = body or {}
+        self.namespace = namespace
+        self.kube_kwargs = kube_kwargs or {}
+        self.kubernetes_api_key_secret = kubernetes_api_key_secret
+
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "body", "namespace", "kube_kwargs", "kubernetes_api_key_secret"
+    )
+    def run(
+        self,
+        body: dict = None,
+        namespace: str = "default",
+        kube_kwargs: dict = None,
+        kubernetes_api_key_secret: str = "KUBERNETES_API_KEY",
+    ) -> None:
+        """
+        Task run method.
+
+        Args:
+            - body (dict, optional): A dictionary representation of a Kubernetes V1Job
+                specification
+            - namespace (str, optional): The Kubernetes namespace to create this job in,
+                defaults to the `default` namespace
+            - kube_kwargs (dict, optional): Optional extra keyword arguments to pass to the
+                Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`)
+            - kubernetes_api_key_secret (str, optional): the name of the Prefect Secret
+                which stored your Kubernetes API Key; this Secret must be a string and in
+                BearerToken format
+
+        Raises:
+            - ValueError: if `body` is `None`
+            - ValueError: if `body["metadata"]["name"] is `None`
+        """
+        if not body:
+            raise ValueError("A dictionary representing a V1Job must be provided.")
+
+        body = {**self.body, **(body or {})}
+        kube_kwargs = {**self.kube_kwargs, **(kube_kwargs or {})}
+
+        job_name = body.get("metadata", {}).get("name")
+        if not job_name:
+            raise ValueError(
+                "The job name must be defined in the body under the metadata key."
+            )
+
+        api_client = cast(
+            client.BatchV1Api, get_kubernetes_client("job", kubernetes_api_key_secret)
+        )
+
+        api_client.create_namespaced_job(namespace=namespace, body=body, **kube_kwargs)
+        self.logger.info(f"Job {job_name} has been created.")
+
+        completed = False
+        while not completed:
+            res = api_client.read_namespaced_job_status(
+                name=job_name, namespace=namespace
+            )
+
+            if res.status.active:
+                time.sleep(self.JOB_STATUS_POOL_INTERVAL)
+            else:
+                if res.status.failed:
+                    raise signals.FAIL(
+                        f"Job {job_name} failed, check Kubernetes pod logs for more information."
+                    )
+                elif res.status.succeeded:
+                    self.logger.info(f"Job {job_name} has been completed.")
+                    completed = True
+
+        api_client.delete_namespaced_job(
```

Comment:
I also think the delete should be toggleable on the task, perhaps with a kwarg that's something like `delete_job_after_completion=True`

Reply:
Full agreement on this. :)